### DISCUSSION TITLE
Is it possible to add rough markers in code blocks?

### DISCUSSION THREAD

**Discussion by senekor** [slidevjs/slidev#1805] [Q&A] [answered]
This would be absolutely amazing, because line highlighting is still a little coarse-grained for me. I'm teaching a language so I need to be able to point out very specific aspects of the syntax.

Using the regular html syntax obviously doesn't work within triple-backticks, it will just be interpreted as raw text. But is it maybe possible to avoid triple-backticks entirely and create code blocks with html syntax only? I wouldn't mind loosing a little bit of the markdown ergonomics.

**Comment by kermanx** [comment, marked as answer]
I think this is possible by adding [Shiki's `transformerMetaWordHighlight`](https://shiki.style/packages/transformers#transformermetawordhighlight) to [`setup/shiki.ts`](https://sli.dev/custom/config-highlighter#configure-shiki), and then use some custom styles.

**Reply by senekor** [reply]
Awesome, that's pretty much exactly what I'm looking for.

I assume this doesn't integrate with the click system at all? So I can't click trough specific underlines? That wouldn't be too bad, I can always duplicate the code blocks and click through those.

Thank you very much.

**Reply by senekor** [reply]
Unfortunately it doesn't seem to work in combination with magic-move either. Not the end of the world, just a limitation I observed.
